### ISSUE TITLE
Deno 1.43.0 fails hard upon encountering lockfile with no `integrity` field.

### ISSUE THREAD

**Issue by nberlette** [denoland/deno#23586]
Attempting to use the REPL on v1.43.0 in a subfolder of a project that I just ran tests on using Deno v1.40.0. I'm greeted with the following error:

```sh
error: Unable to parse contents of lockfile. /Users/nb/work/.../deno.lock: missing field `integrity`
```

Seems rather brittle and backwards-incompatible :/

**Comment by bartlomieju** [comment]
Thanks for the report. How did you get that Deno version? Did you compile from source?

**Comment by dsherret** [comment]
@nberlette was the lockfile generated with jsr specifiers several months ago? That was a purposeful breaking change because JSR specifiers hadn't really landed yet.

**Comment by dsherret** [comment]
Ah, I just re-read the details. Yeah, that's just because JSR specifiers were in 1.40, but still under development. I wouldn't recommend using jsr specifiers with an older deno version.

**Comment by nberlette** [comment]
@bartlomieju I just ran `deno upgrade --version 1.40.0` so I could temporarily benchmark some differences in decorator behavior between the current version and the version they first landed in. 

@dsherret I didn't even think about that aspect that 1.40.0 landed prior to the JSR launch... thank you for pointing that out! Funnily enough, the lockfile didn't even _exist_ before the tests I ran, and it actually never should have as the deno.json file has `lock: false` in it. For some reason the older version generated one anyways, and that's how I stumbled upon this issue. I haven't inspected the file, but it almost certainly would have JSR specifiers in it.
